src/home_robot/home_robot/agent/objectnav_agent/objectnav_agent.py

```diff
@@ -285,16 +291,24 @@ def _preprocess_obs(self, obs: Observations):
         ).unsqueeze(0)
         self.last_poses[0] = curr_pose
         object_goal_category = None
-        if obs.task_observations["object_goal"] is not None:
+        if (
+            "object_goal" in obs.task_observations
+            and obs.task_observations["object_goal"] is not None
+        ):
             object_goal_category = torch.tensor(
                 obs.task_observations["object_goal"]
             ).unsqueeze(0)
         recep_goal_category = None
-        if obs.task_observations["recep_goal"] is not None:
+        if (
+            "recep_goal" in obs.task_observations
+            and obs.task_observations["recep_goal"] is not None
+        ):
             recep_goal_category = torch.tensor(
                 obs.task_observations["recep_goal"]
             ).unsqueeze(0)
         goal_name = [obs.task_observations["goal_name"]]
+        print("object goal =", obs.task_observations["object_goal"])
+        print("recep goal =", obs.task_observations["recep_goal"])
```

Comment:
Both of these will fault when these keys are not present. Since you are anyways checking for those keys above, move these under there?